website/docs/docs/build/derived-metrics.md

```diff
@@ -85,17 +86,97 @@ metrics:
 
 ## Derived metric offset
 
-You may want to use an offset value of a metric in the definition of a derived metric. For example, you can model the retention rate by using a derived metric with an offset, which involves calculating (active customers at the end of the month/active customers at the beginning of the month).
+To perform calculations using a metric's value from a previous time period, you can add an offset parameter to a derived metric. For example, if you want to calculate period-over-period growth or track user retention, you can use this metric offset.
+
+**Note:** You must include the [`metric_time` dimension](/docs/build/dimensions#time) when querying a derived metric with an offset window.
+
+The following example displays how you can calculate monthly revenue growth using a 1-month offset window:
 
 ```yaml
-metrics:
-- name: user_retention
-  type: derived
-  type_params:
-    expr: active_customers/active_customers_t1m
-    metrics:
-      - name: active_customers # these are all metrics (can be a derived metric, meaning building a derived metric with derived metrics)
-      - name: active_customers
-        offset_window: 1 month
-        alias: active_customers_t1m
+- name: customer_retention
+    description: "Percentage of customers that are active now and those active 1 month ago"
+    label: customer_retention
+    type_params:
+      expr: (active_customers/ active_customers_prev_month)
+      metrics:
+        - name: active_customers
+          alias: current_active_customers
+        - name: active_customers
+          offset_window: 1 month
+          alias: active_customers_prev_month
+```
+
+### Offset windows and granularity
+
+You can query any granularity and offset window combination. The following examples queries a metric with a 7-day offset and a monthly grain:
+
+```yaml
+- name: d7_booking_change
+    description: "Difference between bookings now and 7 days ago"
+    type: derived
+    label: d7 Bookings Change
+    type_params:
+      expr: bookings - bookings_7_days_ago
+      metrics:
+        - name: bookings
+          alias: current_bookings
+        - name: bookings
+          offset_window: 7 days
+          alias: bookings_7_days_ago
+
+```
+
+**Using `offset_to_grain`**
```

Comment:
added offset_to_grain based on the [slack convo](https://dbt-labs.slack.com/archives/C050Z6UP4RK/p1694036670014879?thread_ts=1694035028.060429&cid=C050Z6UP4RK) but would like technical verification please? i think it would def be beneficial to add an example, even if it's not as common.

Reply:
Per my earlier comment lets leave this out for now until we can add some examples.